common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -591,6 +591,11 @@ boolean scavengeSome() {
         }
 
         void push(DefaultHandle<?> item) {
+            if (dropHandle(item)) {
```

Comment:
This is only thread-safe if `threadRef.get() == currentThread`, so this will not work. for the `pushLater` case we need to use a count that is kept in the `WeakOrderQueue`.